include/mbedtls/build_info.h

```diff
@@ -74,6 +74,10 @@
 #define MBEDTLS_ARCH_IS_X86
 #endif
 
+#if defined(_M_ARM64) || defined(_M_ARM64EC)
+#define MBEDTLS_PLATFORM_IS_WINDOWS_ON_ARM64
```

Comment:
We should be careful gating any `asm()` blocks on this, as the ABI is different, and certain AArch64 registers cannot be used, as they are not saved during context switches - see "blocked registers" at https://learn.microsoft.com/en-us/windows/arm/arm64ec-abi

Reply:
I don't think we have any MSVC asm - this would only affect MSVC-with-clang (which I would expect is probably not working anyway)